### ISSUE TITLE
always yield full method call to before/after/wrap/etc

### ISSUE THREAD

**Issue by jbodah** [jbodah/spy_rb#22]
Right now we can do cool things like:

```rb
Spy.on(HelloWorld, :hello).before { |context, *args| puts args }
```

It would be really nice to do the following:

```rb
Spy.on(HelloWorld, :hello).after { |mc| puts mc.return_value }
```

**Comment by jbodah** [comment]
Also added for Spy::Instance#when

**Comment by jbodah** [comment]
Included in 1.0.0
